#!/usr/bin/env python3

# Copyright 2022 INRAE, French National Research Institute for Agriculture, Food and Environment
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.


from scout_bringup import urdf_description
from romea_common_bringup import robot_urdf_prefix, robot_prefix
import sys

if __name__ == "__main__":

    argv = sys.argv

    parameters = {}
    for argument in argv[1:]:
        name, value = argument.split(":")
        parameters[name] = value

    mode = parameters["mode"]
    base_name = parameters["base_name"]
    robot_model = parameters["robot_model"]
    prefix = robot_urdf_prefix(parameters["robot_namespace"])
    ros_prefix = robot_prefix(parameters["robot_namespace"])
    print(urdf_description(prefix, mode, base_name, robot_model, ros_prefix))
